contracts/seed/Seed.sol

```diff
@@ -187,14 +204,48 @@ contract Seed {
      * @param _individualCap      The personal cap of each contributor in this class.
      * @param _price              The token price for the addresses in this clas.
      * @param _vestingDuration    The vesting duration for this contributors class.
+     * @param _classVestingStartTime The class vesting start time for the contributor class.
+     * @param _classFee           The fee for the contributor class.
      */
     function addClass(
         uint256 _classCap,
         uint256 _individualCap,
         uint256 _price,
-        uint256 _vestingDuration
+        uint256 _vestingDuration,
+        uint256 _classVestingStartTime,
+        uint256 _classFee
     ) onlyAdmin public {
-        classes.push(ContributorClass(_classCap, _individualCap, _price, _vestingDuration, 0));
+        require(
+            endTime < _classVestingStartTime,
+            "Seed: vesting start time can't be less than endTime"
+        );
+        uint256 seedRequired = (_classCap * PRECISION) / _price;
+        classes.push( ContributorClass(
+                    _classCap,
+                    _individualCap,
+                    _price,
+                    _vestingDuration,
+                    _classVestingStartTime,
+                    _classFee,
+                    0,
+                    seedRequired,
+                    (seedRequired * _classFee) / PRECISION ));
+    }
+
+    /**
+     * @dev                       Set contributor class.
+     * @param _address            Address of the contributor.
+     * @param _class              Class of the contributor.
+     */
+    function setClass(
+        address _address,
+        uint8 _class
+    ) onlyAdmin public {
+        require(_class < classes.length, "Seed: incorrect class chosen");
+        require(!closed, "Seed: should not be closed");
```

Comment:
Btw, I think we need to put one more require here, current_time < startTime, 
Because just setting class might cause problems, for example, we have a contributor who was in a class with personal limit 20 ether, he staked 20 ether. Then we move him to the class with personal limit 1 ether, and class limit 10 ether, what should we do with our contributor then?
So it should be limited

Reply:
I meant not vestingStartTime, but startTime(the value we set in the constructor)